### PR TITLE
chore(cbc): Auto-renew parameter is support update for prepaid resources

### PR DESCRIPTION
**Pull request by Lance52259**
**What this PR does / why we need it**:
Auto-update now supports updates for resources expect EIP and RDS read replica instance.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. Auto-renew parameter is support update for prepaid resources expect EIP and RDS read replica instance.
2. update related tests.
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/cbr' TESTARGS='-run=TestAccVault_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/cbr -v -run=TestAccVault_ -timeout 360m -parallel 4
=== RUN   TestAccVault_BasicServer
=== PAUSE TestAccVault_BasicServer
=== RUN   TestAccVault_ReplicaServer
=== PAUSE TestAccVault_ReplicaServer
=== RUN   TestAccVault_prePaidServer
=== PAUSE TestAccVault_prePaidServer
=== RUN   TestAccVault_BasicVolume
=== PAUSE TestAccVault_BasicVolume
=== RUN   TestAccVault_BasicTurbo
=== PAUSE TestAccVault_BasicTurbo
=== RUN   TestAccVault_ReplicaTurbo
=== PAUSE TestAccVault_ReplicaTurbo
=== RUN   TestAccVault_AutoBind
=== PAUSE TestAccVault_AutoBind
=== CONT  TestAccVault_BasicServer
=== CONT  TestAccVault_BasicTurbo
=== CONT  TestAccVault_AutoBind
=== CONT  TestAccVault_BasicVolume
--- PASS: TestAccVault_AutoBind (26.42s)
=== CONT  TestAccVault_ReplicaTurbo
--- PASS: TestAccVault_ReplicaTurbo (17.28s)
=== CONT  TestAccVault_prePaidServer
--- PASS: TestAccVault_BasicVolume (418.66s)
=== CONT  TestAccVault_ReplicaServer
--- PASS: TestAccVault_ReplicaServer (17.56s)
--- PASS: TestAccVault_BasicServer (472.46s)
--- PASS: TestAccVault_prePaidServer (533.99s)
--- PASS: TestAccVault_BasicTurbo (842.98s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cbr    843.087s
```
```
make testacc TEST='./huaweicloud/services/acceptance/cbr' TESTARGS='-run=TestAccVault_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/cbr -v -run=TestAccVault_ -timeout 360m -parallel 4
=== RUN   TestAccVault_BasicServer
=== PAUSE TestAccVault_BasicServer
=== RUN   TestAccVault_ReplicaServer
=== PAUSE TestAccVault_ReplicaServer
=== RUN   TestAccVault_prePaidServer
=== PAUSE TestAccVault_prePaidServer
=== RUN   TestAccVault_BasicVolume
=== PAUSE TestAccVault_BasicVolume
=== RUN   TestAccVault_BasicTurbo
=== PAUSE TestAccVault_BasicTurbo
=== RUN   TestAccVault_ReplicaTurbo
=== PAUSE TestAccVault_ReplicaTurbo
=== RUN   TestAccVault_AutoBind
=== PAUSE TestAccVault_AutoBind
=== CONT  TestAccVault_BasicServer
=== CONT  TestAccVault_BasicTurbo
=== CONT  TestAccVault_AutoBind
=== CONT  TestAccVault_prePaidServer
--- PASS: TestAccVault_AutoBind (26.09s)
=== CONT  TestAccVault_BasicVolume
--- PASS: TestAccVault_BasicVolume (252.42s)
=== CONT  TestAccVault_ReplicaTurbo
--- PASS: TestAccVault_ReplicaTurbo (16.84s)
=== CONT  TestAccVault_ReplicaServer
--- PASS: TestAccVault_BasicServer (296.96s)
--- PASS: TestAccVault_ReplicaServer (17.33s)
--- PASS: TestAccVault_prePaidServer (403.77s)
--- PASS: TestAccVault_BasicTurbo (803.77s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cbr       803.847s
```
```
make testacc TEST='./huaweicloud/services/acceptance/cce' TESTARGS='-run=TestAccCCEClusterV3_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/cce -v -run=TestAccCCEClusterV3_ -timeout 360m -parallel 4
=== RUN   TestAccCCEClusterV3_basic
=== PAUSE TestAccCCEClusterV3_basic
=== RUN   TestAccCCEClusterV3_prePaid
=== PAUSE TestAccCCEClusterV3_prePaid
=== RUN   TestAccCCEClusterV3_withEip
=== PAUSE TestAccCCEClusterV3_withEip
=== RUN   TestAccCCEClusterV3_withEpsId
=== PAUSE TestAccCCEClusterV3_withEpsId
=== RUN   TestAccCCEClusterV3_turbo
=== PAUSE TestAccCCEClusterV3_turbo
=== RUN   TestAccCCEClusterV3_HibernateAndAwake
=== PAUSE TestAccCCEClusterV3_HibernateAndAwake
=== CONT  TestAccCCEClusterV3_basic
=== CONT  TestAccCCEClusterV3_withEpsId
=== CONT  TestAccCCEClusterV3_withEip
=== CONT  TestAccCCEClusterV3_prePaid
--- PASS: TestAccCCEClusterV3_withEpsId (520.96s)
=== CONT  TestAccCCEClusterV3_HibernateAndAwake
--- PASS: TestAccCCEClusterV3_basic (549.34s)
=== CONT  TestAccCCEClusterV3_turbo
--- PASS: TestAccCCEClusterV3_withEip (559.12s)
--- PASS: TestAccCCEClusterV3_prePaid (684.04s)
--- PASS: TestAccCCEClusterV3_turbo (560.56s)
--- PASS: TestAccCCEClusterV3_HibernateAndAwake (1234.78s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cce     1755.815s
```
```
make testacc TEST='./huaweicloud/services/acceptance/css' TESTARGS='-run=TestAccCssCluster_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/css -v -run=TestAccCssCluster_ -timeout 360m -parallel 4
=== RUN   TestAccCssCluster_basic
=== PAUSE TestAccCssCluster_basic
=== RUN   TestAccCssCluster_prePaid
=== PAUSE TestAccCssCluster_prePaid
=== CONT  TestAccCssCluster_basic
=== CONT  TestAccCssCluster_prePaid
--- PASS: TestAccCssCluster_prePaid (1278.55s)
--- PASS: TestAccCssCluster_basic (1791.91s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/css 1791.984s
```
```
make testacc TEST='./huaweicloud/services/acceptance/dcs' TESTARGS='-run=TestAccDcsInstances_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dcs -v -run=TestAccDcsInstances_ -timeout 360m -parallel 4
=== RUN   TestAccDcsInstances_basic
=== PAUSE TestAccDcsInstances_basic
=== RUN   TestAccDcsInstances_withEpsId
=== PAUSE TestAccDcsInstances_withEpsId
=== RUN   TestAccDcsInstances_whitelists
=== PAUSE TestAccDcsInstances_whitelists
=== RUN   TestAccDcsInstances_tiny
=== PAUSE TestAccDcsInstances_tiny
=== RUN   TestAccDcsInstances_single
=== PAUSE TestAccDcsInstances_single
=== RUN   TestAccDcsInstances_prePaid
=== PAUSE TestAccDcsInstances_prePaid
=== CONT  TestAccDcsInstances_basic
=== CONT  TestAccDcsInstances_tiny
=== CONT  TestAccDcsInstances_prePaid
=== CONT  TestAccDcsInstances_single
--- PASS: TestAccDcsInstances_single (73.14s)
=== CONT  TestAccDcsInstances_whitelists
--- PASS: TestAccDcsInstances_tiny (84.86s)
=== CONT  TestAccDcsInstances_withEpsId
--- PASS: TestAccDcsInstances_basic (118.53s)
--- PASS: TestAccDcsInstances_whitelists (81.51s)
--- PASS: TestAccDcsInstances_withEpsId (70.23s)
--- PASS: TestAccDcsInstances_prePaid (248.97s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dcs        249.094s
```
```
make testacc TEST='./huaweicloud/services/acceptance/dds' TESTARGS='-run=TestAccDDSV3Instance_prePaid'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dds -v -run=TestAccDDSV3Instance_prePaid -timeout 360m -parallel 4
=== RUN   TestAccDDSV3Instance_prePaid
=== PAUSE TestAccDDSV3Instance_prePaid
=== CONT  TestAccDDSV3Instance_prePaid
--- PASS: TestAccDDSV3Instance_prePaid (1641.11s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dds     1641.182s
```
```
make testacc TEST='./huaweicloud/services/acceptance/elb' TESTARGS='-run=TestAccElb
V3LoadBalancer_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/elb -v -run=TestAccElbV3LoadBalancer_ -timeout 360m -parallel 4
=== RUN   TestAccElbV3LoadBalancer_basic
=== PAUSE TestAccElbV3LoadBalancer_basic
=== RUN   TestAccElbV3LoadBalancer_withEpsId
=== PAUSE TestAccElbV3LoadBalancer_withEpsId
=== RUN   TestAccElbV3LoadBalancer_withEIP
=== PAUSE TestAccElbV3LoadBalancer_withEIP
=== RUN   TestAccElbV3LoadBalancer_prePaid
=== PAUSE TestAccElbV3LoadBalancer_prePaid
=== CONT  TestAccElbV3LoadBalancer_basic
=== CONT  TestAccElbV3LoadBalancer_withEIP
=== CONT  TestAccElbV3LoadBalancer_withEpsId
=== CONT  TestAccElbV3LoadBalancer_prePaid
--- PASS: TestAccElbV3LoadBalancer_withEpsId (29.54s)
--- PASS: TestAccElbV3LoadBalancer_withEIP (32.10s)
--- PASS: TestAccElbV3LoadBalancer_basic (57.14s)
--- PASS: TestAccElbV3LoadBalancer_prePaid (135.79s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/elb    135.882s
```
```
make testacc TEST='./huaweicloud/services/acceptance/evs' TESTARGS='-run=TestAccEvsVolume_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/evs -v -run=TestAccEvsVolume_ -timeout 360m -parallel 4
=== RUN   TestAccEvsVolume_basic
=== PAUSE TestAccEvsVolume_basic
=== RUN   TestAccEvsVolume_withEpsId
=== PAUSE TestAccEvsVolume_withEpsId
=== RUN   TestAccEvsVolume_prePaid
=== PAUSE TestAccEvsVolume_prePaid
=== CONT  TestAccEvsVolume_basic
=== CONT  TestAccEvsVolume_prePaid
=== CONT  TestAccEvsVolume_withEpsId
--- PASS: TestAccEvsVolume_withEpsId (41.26s)
--- PASS: TestAccEvsVolume_basic (83.99s)
--- PASS: TestAccEvsVolume_prePaid (92.19s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/evs       92.264s
```
```
make testacc TEST='./huaweicloud/services/acceptance/gaussdb' TESTARGS='-run=TestAccGeminiDBInstance_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/gaussdb -v -run=TestAccGeminiDBInstance_ -timeout 360m -parallel 4
=== RUN   TestAccGeminiDBInstance_basic
=== PAUSE TestAccGeminiDBInstance_basic
=== RUN   TestAccGeminiDBInstance_prePaid
=== PAUSE TestAccGeminiDBInstance_prePaid
=== CONT  TestAccGeminiDBInstance_basic
=== CONT  TestAccGeminiDBInstance_prePaid
--- PASS: TestAccGeminiDBInstance_basic (1068.01s)
--- PASS: TestAccGeminiDBInstance_prePaid (1155.94s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/gaussdb   1156.017s
```
```
make testacc TEST='./huaweicloud/services/acceptance/gaussdb' TESTARGS='-run=TestAccGaussInfluxInstance_prePaid'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/gaussdb -v -run=TestAccGaussInfluxInstance_prePaid -timeout 360m -parallel 4
=== RUN   TestAccGaussInfluxInstance_prePaid
=== PAUSE TestAccGaussInfluxInstance_prePaid
=== CONT  TestAccGaussInfluxInstance_prePaid
--- PASS: TestAccGaussInfluxInstance_prePaid (1215.98s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/gaussdb   1216.058s
```
```
make testacc TEST='./huaweicloud/services/acceptance/gaussdb' TESTARGS='-run=TestAccGaussMongoInstance_prePaid'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/gaussdb -v -run=TestAccGaussMongoInstance_prePaid -timeout 360m -parallel 4
=== RUN   TestAccGaussMongoInstance_prePaid
=== PAUSE TestAccGaussMongoInstance_prePaid
=== CONT  TestAccGaussMongoInstance_prePaid
--- PASS: TestAccGaussMongoInstance_prePaid (1196.38s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/gaussdb   1196.461s
```
```
make testacc TEST='./huaweicloud/services/acceptance/gaussdb' TESTARGS='-run=TestAccGaussDBInstance_prePaid'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/gaussdb -v -run=TestAccGaussDBInstance_prePaid -timeout 360m -parallel 4
=== RUN   TestAccGaussDBInstance_prePaid
=== PAUSE TestAccGaussDBInstance_prePaid
=== CONT  TestAccGaussDBInstance_prePaid
--- PASS: TestAccGaussDBInstance_prePaid (1303.86s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/gaussdb   1303.946s
```
```
make testacc TEST='./huaweicloud/services/acceptance/rds' TESTARGS='-run=TestAccRds
Instance_prePaid'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/rds -v -run=TestAccRdsInstance_prePaid -timeout 360m -parallel 4
=== RUN   TestAccRdsInstance_prePaid
=== PAUSE TestAccRdsInstance_prePaid
=== CONT  TestAccRdsInstance_prePaid
--- PASS: TestAccRdsInstance_prePaid (1509.97s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/rds        1510.234s
```

### CODE REVIEW

**Review by iskey — APPROVED** [review]

**Comment by Lance52259** [comment]
Tests for CCE services are now passable.

**Comment by Lance52259** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/cce' TESTARGS='-run=TestAccCCENodeV3_prePaid'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/cce -v -run=TestAccCCENodeV3_prePaid -timeout 360m -parallel 4
=== RUN   TestAccCCENodeV3_prePaid
=== PAUSE TestAccCCENodeV3_prePaid
=== CONT  TestAccCCENodeV3_prePaid
--- PASS: TestAccCCENodeV3_prePaid (1125.57s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cce       1125.648s
```

File: huaweicloud/services/cce/resource_huaweicloud_cce_cluster_v3.go [hunk 1]
```diff
@@ -524,7 +541,7 @@ func resourceCCEClusterV3Create(ctx context.Context, d *schema.ResourceData, met
 	if d.Get("hibernate").(bool) {
 		err = resourceCCEClusterV3Hibernate(ctx, d, cceClient)
 		if err != nil {
-			diags = append(diags, diag.FromErr(err)...)
+			diags = append(diags, fmtp.DiagErrorf("Error installing ICAgent in CCE cluster: %s", err)[0])
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/cce/resource_huaweicloud_cce_cluster_v3.go]: error message not correct

File: huaweicloud/services/cce/resource_huaweicloud_cce_node_v3.go [hunk 2]
```diff
@@ -817,7 +836,7 @@ func resourceCCENodeV3Create(ctx context.Context, d *schema.ResourceData, meta i
 	stateConf := &resource.StateChangeConf{
 		Pending:      []string{"Build", "Installing"},
 		Target:       []string{"Active"},
-		Refresh:      waitForCceNodeActive(nodeClient, clusterid, nodeID),
+		Refresh:      waitForCceNodeActive(nodeClient, clusterid, d.Id()),
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/cce/resource_huaweicloud_cce_node_v3.go]: replace d.Id() with nodeID

File: huaweicloud/services/elb/resource_huaweicloud_elb_loadbalancer.go [hunk 3]
```diff
@@ -448,11 +453,10 @@ func resourceLoadBalancerV3Update(ctx context.Context, d *schema.ResourceData, m
...
 		if d.Get("charging_mode").(string) == "prePaid" && d.HasChanges("l4_flavor_id", "l7_flavor_id") {
-			autoRenew, _ := strconv.ParseBool(d.Get("auto_renew").(string))
 			prepaidOpts := loadbalancers.PrepaidOpts{
 				PeriodType: d.Get("period_unit").(string),
 				PeriodNum:  d.Get("period").(int),
-				AutoRenew:  autoRenew,
+				AutoPay:    true,
 			}
 			if d.Get("auto_pay").(string) != "false" {
 				prepaidOpts.AutoPay = true
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/elb/resource_huaweicloud_elb_loadbalancer.go]: duplict auto_pay

Comment (Lance52259) [reply huaweicloud/services/elb/resource_huaweicloud_elb_loadbalancer.go]: done.

File: huaweicloud/services/cce/resource_huaweicloud_cce_node_v3.go (hunk 2 above)
Comment (Lance52259) [reply huaweicloud/services/cce/resource_huaweicloud_cce_node_v3.go]: done.

File: huaweicloud/services/cce/resource_huaweicloud_cce_cluster_v3.go (hunk 1 above)
Comment (Lance52259) [reply huaweicloud/services/cce/resource_huaweicloud_cce_cluster_v3.go]: done.

**Comment by Lance52259** [comment]
```
make testacc TEST='./huaweicloud' TESTARGS='-run=TestAccComputeInstance_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud -v -run=TestAccComputeInstance_ -timeout 360m -parallel 4
=== RUN   TestAccComputeInstance_basic
=== PAUSE TestAccComputeInstance_basic
=== RUN   TestAccComputeInstance_disks
=== PAUSE TestAccComputeInstance_disks
=== RUN   TestAccComputeInstance_prePaid
=== PAUSE TestAccComputeInstance_prePaid
=== RUN   TestAccComputeInstance_tags
=== PAUSE TestAccComputeInstance_tags
=== RUN   TestAccComputeInstance_powerAction
=== PAUSE TestAccComputeInstance_powerAction
=== CONT  TestAccComputeInstance_basic
=== CONT  TestAccComputeInstance_tags
=== CONT  TestAccComputeInstance_prePaid
=== CONT  TestAccComputeInstance_disks
--- PASS: TestAccComputeInstance_basic (204.39s)
=== CONT  TestAccComputeInstance_powerAction
--- PASS: TestAccComputeInstance_prePaid (219.15s)
--- PASS: TestAccComputeInstance_disks (230.85s)
--- PASS: TestAccComputeInstance_tags (266.12s)
--- PASS: TestAccComputeInstance_powerAction (474.02s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud       678.507s
```

File: huaweicloud/services/cce/resource_huaweicloud_cce_node_v3.go [hunk 4]
```diff
@@ -947,9 +966,48 @@ func resourceCCENodeV3Update(ctx context.Context, d *schema.ResourceData, meta i
 		}
 	}
 
+	if d.HasChange("auto_renew") {
+		bssClient, err := config.BssV2Client(config.GetRegion(d))
+		if err != nil {
+			return diag.Errorf("error creating BSS V2 client: %s", err)
+		}
+		ecsId, err := getUnderlyingResourceId(nodeClient, bssClient, d.Get("cluster_id").(string), d.Id())
```
Comment (ShiChangkuo) [inline huaweicloud/services/cce/resource_huaweicloud_cce_node_v3.go]: can you try to use `server_id` attribute?

File: huaweicloud/services/dcs/resource_huaweicloud_dcs_instance.go [hunk 5]
```diff
@@ -586,6 +586,14 @@ func resourceDcsInstancesCreate(ctx context.Context, d *schema.ResourceData, met
 			return fmtp.DiagErrorf("[DEBUG] Error the order is not completed while "+
 				"creating DCS instance. %s : %#v", d.Id(), err)
 		}
+		err = common.WaitOrderComplete(ctx, bssClient, r.OrderId, d.Timeout(schema.TimeoutCreate))
+		if err != nil {
+			return diag.Errorf("the order is not completed while creating CBR vault: %v", err)
```
Comment (ShiChangkuo) [inline huaweicloud/services/dcs/resource_huaweicloud_dcs_instance.go]: the error message is  not correct

Comment (Lance52259) [reply huaweicloud/services/dcs/resource_huaweicloud_dcs_instance.go]: all incorrect message have been modified.

File: huaweicloud/services/cce/resource_huaweicloud_cce_node_v3.go (hunk 4 above)
Comment (Lance52259) [reply huaweicloud/services/cce/resource_huaweicloud_cce_node_v3.go]: it's effective.

**Pull request merged**